src/sdk/namespace_fs.js

```diff
@@ -924,6 +919,24 @@ class NamespaceFS {
         }
     }
 
+    async validate_results(results, fs_context) {
+        await Promise.all(results.map(async r => {
+            if (r.common_prefix) return;
+            const entry_path = path.join(this.bucket_path, r.key);
+            //If entry is outside of bucket, returns stat of symbolic link
+            const use_lstat = !(await this._is_path_in_bucket_boundaries(fs_context, entry_path));
+            try {
+                r.stat = await nb_native().fs.stat(fs_context, entry_path, { use_lstat });
+            } catch (err) {
+                dbg.warn('NamespaceFS: validate_results : couldnt access file entry_path', entry_path, ", skipping...");
+                const index = results.indexOf(r);
+                if (index > -1) {
+                    results.splice(index, 1);
```

Comment:
what about bucket owner, do we always want to remove this entry from the list? this should be compatible to S3

Reply:
@romayalon When removing an item from the list, there is a chance the list object returns with less than 1000 items One solution would be to check the access while adding the item to result in an array and return 1000 items